### PR TITLE
Add get user image to the test suite. 

### PR DESCRIPTION
**Pull request by aaujayasena**
Add new script for get user image 
Add data creation script post user image
Add artifacts for the test, image with 1MB size and .csv file for record images IDs.
Update the perf-test.sh and run-performance-tests
Define nginx worker nodes
Add 5.9.0_two-node_4-core test results

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,11 @@ These performance test scripts make use of the Apache JMeter to run the tests wi
 
 The deployment is automated using AWS cloud formation. 
 
-Artifacts in the master branch can be used with the WSO2 Identity Server version 5.9.0 and onwards. If you wish to benchmark previous releases, please use the branch [is-5.8.0](https://github.com/wso2/performance-is/tree/is-5.8.0)
-
+Artifacts in the master branch can be used with the WSO2 Identity Server version 5.10.0. If you wish to
+ benchmark previous releases, please use the branch, 
+ 1. [is-5.9.0](https://github.com/wso2/performance-is/tree/is-5.9.0) for product version 5.9.0
+ 2. [is-5.8.0](https://github.com/wso2/performance-is/tree/is-5.8.0) for  product version 5.8.0 to 5.6.0
```
Comment (vihanga-liyanage) [inline README.md]: There's an additional space.

File: common/jmeter/perf-test-is.sh [hunk 2]
```diff
@@ -330,7 +330,7 @@ function run_test_data_scripts() {
 
     echo "Running test data setup scripts"
     echo "=========================================================================================="
-    declare -a scripts=("TestData_SCIM2_Add_User.jmx" "TestData_Add_OAuth_Apps.jmx" "TestData_Add_SAML_Apps.jmx")
+    declare -a scripts=("TestData_SCIM2_Add_User.jmx" "TestData_Add_OAuth_Apps.jmx" "TestData_Add_SAML_Apps.jmx" "Post_image.jmx")
```
Comment (vihanga-liyanage) [inline common/jmeter/perf-test-is.sh]: Can we rename the data script according to the other data adding scripts, please? Suggestion: **TestData_Add_images**

File: common/jmeter/setup/Post_image.jmx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+          <elementProp name="adminCredentials" elementType="Argument">
+            <stringProp name="Argument.name">adminCredentials</stringProp>
+            <stringProp name="Argument.value">YWRtaW46YWRtaW4=</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+            <stringProp name="Argument.desc">admin:admin</stringProp>
+          </elementProp>
+        </collectionProp>
+        <stringProp name="TestPlan.comments">ec2-34-207-59-213.compute-1.amazonaws.com</stringProp>
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: Is this necessary?

File: common/jmeter/setup/Post_image.jmx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+            <stringProp name="Argument.value">password</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+          <elementProp name="rolename" elementType="Argument">
+            <stringProp name="Argument.name">rolename</stringProp>
+            <stringProp name="Argument.value">isTestUserRole</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+        </collectionProp>
+        <stringProp name="TestPlan.comments">ec2-34-207-59-213.compute-1.amazonaws.com</stringProp>
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: Is this necessary?

File: common/jmeter/setup/Post_image.jmx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+            <stringProp name="RegexExtractor.refname">location</stringProp>
+            <stringProp name="RegexExtractor.regex">Location: (.*)</stringProp>
+            <stringProp name="RegexExtractor.template">$1$</stringProp>
+            <stringProp name="RegexExtractor.default"></stringProp>
+            <stringProp name="RegexExtractor.match_number">1</stringProp>
+            <stringProp name="Sample.scope">all</stringProp>
+          </RegexExtractor>
+          <hashTree/>
+        </hashTree>
+        <ResultCollector guiclass="ViewResultsFullVisualizer" testclass="ResultCollector" testname="View Results Tree" enabled="true">
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: Result tree should be disabled by default.

File: common/jmeter/setup/Post_image.jmx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+              <threadCounts>true</threadCounts>
+              <idleTime>true</idleTime>
+              <connectTime>true</connectTime>
+            </value>
+          </objProp>
+          <stringProp name="filename"></stringProp>
+        </ResultCollector>
+        <hashTree/>
+      </hashTree>
+      <ResultCollector guiclass="SummaryReport" testclass="ResultCollector" testname="Summary Report" enabled="true">
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: The summary report should be disabled by default.

File: common/jmeter/setup/Post_image.jmx [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+          <elementProp name="rampUpPeriod" elementType="Argument">
+            <stringProp name="Argument.name">rampUpPeriod</stringProp>
+            <stringProp name="Argument.value">${__P(rampUpPeriod,10)}</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+          <elementProp name="locationCsvPath" elementType="Argument">
+            <stringProp name="Argument.name">locationCsvPath</stringProp>
+            <stringProp name="Argument.value">/home/ubuntu/location.csv</stringProp>
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: Rename this file to a self-explanatory name. Suggestion: **image_service_test_data.csv**

File: common/jmeter/setup/Post_image.jmx [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
+          <elementProp name="imageLocationPath" elementType="Argument">
+            <stringProp name="Argument.name">imageLocationPath</stringProp>
+            <stringProp name="Argument.value">/home/ubuntu/image.png</stringProp>
```
Comment (vihanga-liyanage) [inline common/jmeter/setup/Post_image.jmx]: Is this path correct?

File: single-node/run-performance-tests.sh [hunk 9]
```diff
@@ -100,6 +100,13 @@ declare -A test_scenario9=(
     [jmx]="saml/SAML2_SSO_Redirect_Binding.jmx"
     [skip]=false
 )
+declare -A test_scenario10=(
+    [name]="10-Get_user_image"
+    [display_name]="Get user images"
+    [description]="Get the image"
```
Comment (vihanga-liyanage) [inline single-node/run-performance-tests.sh]: Suggestion: **Testing the GET request of the image API**

File: two-node-cluster/setup/setup-nginx.sh [hunk 10]
```diff
@@ -74,4 +74,9 @@ echo "============================================"
 sudo sed -i 's$server xxx.xxx.xxx.1:9443$server '$wso2_is_1_ip':9443$g' /etc/nginx/conf.d/is.conf || echo "error 1"
 sudo sed -i 's$server xxx.xxx.xxx.2:9443$server '$wso2_is_2_ip':9443$g' /etc/nginx/conf.d/is.conf || echo "error 1"
 
+echo ""
+echo "Adding workerconnection to nginx.conf file"
+echo "============================================"
+sudo sed -i 's/worker_connections 768/worker_connections 1500/g' /etc/nginx/nginx.conf || echo "error 1"
```
Comment (vihanga-liyanage) [inline two-node-cluster/setup/setup-nginx.sh]: Why do we need this?

File: common/jmeter/setup/Post_image.jmx (hunk 8 above)
Comment (aaujayasena) [reply common/jmeter/setup/Post_image.jmx]: yes path is correct.

File: two-node-cluster/setup/setup-nginx.sh (hunk 10 above)
Comment (aaujayasena) [reply two-node-cluster/setup/setup-nginx.sh]: nginx throw 'worker_connections are not enough' alert for two node and 4 core perfroamnce test execution with 500 concurrency . Default worker connection number is 512 and currently using the default one.

**Comment by aaujayasena** [comment]
Update the files according to comments . [1]
[1] https://github.com/wso2/performance-is/pull/63/commits/f7eee15f4b8e67ebfd68d48c69c1ae789ab46dc1

File: common/jmeter/setup/Post_image.jmx (hunk 8 above)
Comment (vihanga-liyanage) [reply common/jmeter/setup/Post_image.jmx]: I noticed that you've uploaded an image as well. Can't we use that? User will have to save an image to /home/ubuntu for this to work right?

Comment (aaujayasena) [reply common/jmeter/setup/Post_image.jmx]: Image file already given, [1].

[1] https://github.com/wso2/performance-is/pull/63/files#diff-c68999ca672d9d5510ec301aab718938

Comment (vihanga-liyanage) [reply common/jmeter/setup/Post_image.jmx]: Yes, my question is, the path /home/ubuntu/image.png is correct or not. I see that you've added the image into common/lib folder but in the script we refer it from /home/ubuntu?

Comment (aaujayasena) [reply common/jmeter/setup/Post_image.jmx]: Image file will be place on the bastion node at the /home/ubuntu path from the set up file.

**Comment by aaujayasena** [comment]
1. Add separate resource folder and moved the resources (image.png/image_service_test_data.csv)  as discussed.
2. Update the deployment.toml file with uniqueid JDBC user store 

https://github.com/wso2/performance-is/pull/63/commits/430dfe40bc68ac38a217078bbc033a098b2efbe2

https://github.com/wso2/performance-is/pull/63/commits/100ff58e966c53f3d27e8eda57629fe6db71497d

File: single-node/setup/resources/deployment.toml [hunk 11]
```diff
@@ -9,11 +9,11 @@ password = "admin"
...
-type = "database"
-connection_url = "ldap://localhost:${Ports.EmbeddedLDAP.LDAPServerPort}"
-connection_name = "uid=admin,ou=system"
-connection_password = "admin"
-base_dn = "dc=wso2,dc=org"
+type = "database_unique_id"
+#connection_url = "ldap://localhost:${Ports.EmbeddedLDAP.LDAPServerPort}"
+##connection_name = "uid=admin,ou=system"
+#connection_password = "admin"
+#base_dn = "dc=wso2,dc=org"
```
Comment (vihanga-liyanage) [inline single-node/setup/resources/deployment.toml:16]: Why do we need to keep these?

Comment (aaujayasena) [reply single-node/setup/resources/deployment.toml:16]: These will be help if we need to run the test in other user stores. For JDBC theses configurations not valid.

Comment (vihanga-liyanage) [reply single-node/setup/resources/deployment.toml:16]: then let's add comments to easily understand that.

Comment (aaujayasena) [reply single-node/setup/resources/deployment.toml:16]: updated. https://github.com/wso2/performance-is/pull/69

**Pull request merged**